epsagon/trace.py

```diff
@@ -887,6 +902,27 @@ def remove_ignored_keys(self, input_dict):
                     copied_dict[key] = self.remove_ignored_keys(value)
         return copied_dict
 
+    def get_dict_with_allow_keys(self, input_dict):
+        """
+        Keeping only the branches which contains allowed keys.
+        Any branch that doesn't contain allowed key will be deleted.
+        :param input_dict: Input dict to remove branches without allowed keys.
+        :return: a dict that contains branches which contains allowed keys
+        """
+        copied_dict = input_dict.copy()
+        for key in input_dict:
+            is_key_in_whitelist = self._strip_key(key) in self.keys_to_allow
```

Comment:
this variable is redundant
you can just:
```
if self._strip_key(key) not in self.keys_to_allow
```

Reply:
I saw that, But I think the function is more readable and clear in this way.